### PR TITLE
Add ability to use redis db instead of json db

### PR DESCRIPTION
**Pull request by ulysseses**
Add ability to use redis db instead of json db

`USE_REDIS=true` env var will use Redis DB instead of JSON (flat-file) db.

`REDIS_PORT` and `REDIS_HOST` configure the Redis instance.

The redis and json DB both have the same "interface". B/c redis works asynchronously, I've transformed the interface to be async, as historically it's been sync.

### CODE REVIEW

**Comment by ulysseses** [comment]
Feel free to merge. I haven't tested it yet (b/c I don't have a redis instance locally (yet). I can try to test but I have to be gone for ~1 hour.

**Review by lrt98802 — APPROVED** [review]

**Comment by lrt98802** [comment]
Oops seems like there are merge conflicts on `package-lock.json` @ulysseses

File: src/db/index.js [hunk 1]
```diff
@@ -1,83 +1,99 @@
...
+require("dotenv").config();
 
 const RESEED_PATH_PREFIX = "/reseed";
 const TRACKER_PATH_PREFIX = "/tracker";
 const SUBMIT_PATH_PREFIX = "/submit";
 
-// Implements concurrent DB supporting thread-safe read/write/delete operations
-class ConcurrentDB {
+// Decide between Redis or JsonDB.
+const USE_REDIS = process.env.USE_REDIS === "true";
```
Comment (lrt98802) [inline src/db/index.js:12]: Will this environment variable be set automatically by Heroku when it starts our app?

Comment (ulysseses) [reply src/db/index.js:15]: No, you'll need to set this env var

**Pull request merged**